### PR TITLE
Added locales support and Broadcast_text support

### LINKED ISSUE #197: Locales
**Description**: Can you show me how I can change locales into Russian lang? (DBC.Locale = 8)

**Current behaviour**: I changed DBC locales in configs and set RealmZone to 12 and nothing happened. I have fully completed locales tables fpt loc 8
**Expected behaviour**: Lang in the game must changes

**Steps to reproduce the problem**:
1. 
2. 
3. 

**Branch(es)**: 0.x / 1.x / master (Specify the branch(es) affected by this issue) 

**AC hash/commit**:  

**Operating system**:  

**Modules**:  Are you using modules? if yes, list them

**Other Customizations**: Are you use core/db patches or other customizations? If yes please specify them here.
## 

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/38620954-locales?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by sucofog**
**Changes proposed:**

I discovered there is a supossed "FIX" that does not work becouse it gives a bunch of errors compiling it under Windows 10, i discovered it due to @cooler-SAI forked it, thanks to him for answering me on his fork (hope he don't gets angry for using his commits to make a pull request), the "FIX" was on this commits on bitbucket platform:

https://bitbucket.org/pak3935/sunwellcore/commits/db58f8ae1a4942622b368840aee474787a528a6f

and

https://bitbucket.org/pak3935/sunwellcore/commits/078136139b6777fcd64e9a4c871c7d58c8e5d5a5

This commit supposedly FIX's two things:
1. Locales support to all locales_ translations on database
2. Broadcast_text and locales_broadcast_text

**Target branch(es):** Master

**Issues addressed:** Closes #197 #207 

**Tests performed:** It does not build, it can't be tested

**Known issues and TODO list:**
- [ locales_XXXXXXXX_XXXXXX tables that translates texts to different languages ]
- [ Broadcast_text and locales_broadcast_text ]

Sorry if i did not explain everything well.

### CODE REVIEW

**Comment by sucofog** [comment]
@ShinDarth @Krath

**Comment by HeartWell** [comment]
It's bad, pak3935 the user cut out all that is not necessary.

**Comment by sucofog** [comment]
What do you mean with "the user cut out all that is not necessary" @HeartWell ?
I did not understand :P

**Comment by HeartWell** [comment]
In simple words he broke more than he did, it was tested by me.

**Comment by sucofog** [comment]
And is there any possible way to solve this?
So this won't work although the code is compatible?

**Comment by HeartWell** [comment]
This will work, but it will lead to other problems of localization.

**Comment by sucofog** [comment]
I mean i think it's a very important part to fix this, so if this code is usable in a possible way to fix locales and broadcast_text..
:(

File: src/game/AI/SmartScripts/SmartScript.cpp [hunk 1]
```diff
@@ -789,21 +789,20 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
             delete targets;
             break;
         }
-        case SMART_ACTION_FLEE_FOR_ASSIST:
-        {
-            // Xinef: do not allow to flee without control (stun, fear etc)
```
Comment (HeartWell) [inline src/game/AI/SmartScripts/SmartScript.cpp]: Why this delete?

File: src/game/Entities/Creature/GossipDef.cpp [hunk 2]
```diff
@@ -22,38 +22,62 @@ GossipMenu::~GossipMenu()
     ClearMenu();
 }
 
-void GossipMenu::AddMenuItem(int32 menuItemId, uint8 icon, std::string const& message, uint32 sender, uint32 action, std::string const& boxMessage, uint32 boxMoney, bool coded /*= false*/)
```
Comment (HeartWell) [inline src/game/Entities/Creature/GossipDef.cpp]: Why this delete?

**Comment by sucofog** [comment]
I don't really know, i mean i posted the pull request becouse i hoped it may be a good fix, but if its not compatible with the code so i didn't tested, but if you did and it did lead to other localization problems then, i don't know what to say, just to hope someone fix's this soon, im a noob programming, the solution i did was in database translating original creature_template, quest_template, and other tables with the esES translations i got, and i did a database backup of the original database, in case azerothcore community brings a definitive solution, hope it happens 👍 
Anyways thanks for your quick response @HeartWell very apreciated

**Comment by cooler-SAI** [comment]
on Sunwell Locales works good ( Tested 100%), not Ideal but all works!.Sunwell Locales commits in azerothcore have errors with compile. I'm not good in C++ but I think someone can help here.

**Comment by HeartWell** [comment]
Again, I pointed to the rows are deleted without thinking.

**Comment by sucofog** [comment]
But how did you compiled it @cooler-SAI ? I tried to do a fresh compile on VS 2015 and i got a lot of errors so i could not compile :/

**Comment by cooler-SAI** [comment]
use only VS2013 for Sunwell compile

**Comment by ShinDarth** [comment]
Just to clarify:

AzerothCore is not exaclty "Mangos + TrinityCore + Sunwellcore".

SunwellCore is based on TrinityCore
TrinityCore is based on MaNGOS

AzerothCore is based on SunwellCore, but we are importing all the good stuff
that was added in CMaNGOS and/or TrinityCore that was not present in SunwellCore

So **whatever thing is working on TrinityCore, can also be imported to AzerothCore**.

Sometimes it is very easy to import stuff from TrinityCore to AzerothCore,
sometimes it requires other things to be imported
(e.g. you try to import X but X **depends on Y** so you have to **import Y first**)

**Comment by Blumfield** [comment]
I'm going to backport locales from TC

**Comment by sucofog** [comment]
That would be wonderful @Blumfield 💃

**Comment by cooler-SAI** [comment]
Good! coz all who create Locales for Sunwell don't wanna shared it ( only use for private)

**Comment by Blumfield** [comment]
See #218

**Comment by HeartWell** [comment]
https://github.com/azerothcore/azerothcore-wotlk/pull/218 dublicate

**Comment by sucofog** [comment]
I close @HeartWell thanks!

**Pull request closed without merging**